code/modules/admin/verbs/map_export.dm

```diff
@@ -0,0 +1,261 @@
+	usr << ftp("data/[file_name].dmm", "[file_name].dmm")
+
+/proc/sanitize_filename(t)
+	return hashtag_newlines_and_tabs(t, list("\n"="", "\t"="", "/"="", "\\"="", "?"="", "%"="", "*"="", ":"="", "|"="", "\""="", "<"="", ">"=""))
+
+/proc/hashtag_newlines_and_tabs(text, list/repl_chars = list("\n"="#","\t"="#"))
+	for(var/char in repl_chars)
+		var/index = findtext(text, char)
+		while(index)
+			text = copytext(text, 1, index) + repl_chars[char] + copytext(text, index + length(char))
+			index = findtext(text, char, index + length(char))
+	return text
+
+/obj/proc/on_object_saved(var/depth = 0)
+	return ""
+
+// Save resources in silo
+/obj/machinery/ore_silo/on_object_saved(var/depth = 0)
+	if(depth >= 10)
+		return ""
+	var/data
+	var/datum/component/material_container/material_holder = GetComponent(/datum/component/material_container)
+	for(var/each in material_holder.materials)
+		var/amount = material_holder.materials[each] / 100
+		var/datum/material/material_datum = each
+		while(amount > 0)
+			var/amount_in_stack = max(1, min(50, amount))
+			amount -= amount_in_stack
+			data += "[data ? ",\n" : ""][material_datum.sheet_type]{\n\tamount = [amount_in_stack]\n\t}"
+	return data
+
+/**Map exporter
+* Inputting a list of turfs into convert_map_to_tgm() will output a string
+* with the turfs and their objects / areas on said turf into the TGM mapping format
+* for .dmm files. This file can then be opened in the map editor or imported
+* back into the game.
+* ============================
+* This has been made semi-modular so you should be able to use these functions
+* elsewhere in code if you ever need to get a file in the .dmm format
+**/
+/atom/proc/get_save_vars()
+	return list("pixel_x",
+				"pixel_y",
+				"dir",
+				"name",
+				"req_access",
+				"piping_layer", 
+				"color", 
+				"icon", 
+				"icon_state", 
+				"pipe_color", 
+				"amount", 
+				"dwidth",
+				"dheight", 
+				"height", 
+				"width", 
+				"roundstart_template", 
+				"shuttle_id"
+			)
+
+GLOBAL_LIST_INIT(save_file_chars, list(
+	"a","b","c","d","e",
+	"f","g","h","i","j",
+	"k","l","m","n","o",
+	"p","q","r","s","t",
+	"u","v","w","x","y",
+	"z","A","B","C","D",
+	"E","F","G","H","I",
+	"J","K","L","M","N",
+	"O","P","Q","R","S",
+	"T","U","V","W","X",
+	"Y","Z"
+))
+
+/proc/to_list_string(list/l)
+	. = "list("
+	var/first_entry = TRUE
+	for(var/item in l)
+		if(!first_entry)
+			. += ", "
+		if(l[item])
+			. += hashtag_newlines_and_tabs("[item] = [l[item]]", list("{"="", "}"="", "\""="", ";"="", ","=""))
+		else
+			. += hashtag_newlines_and_tabs("[item]", list("{"="", "}"="", "\""="", ";"="", ","=""))
+		first_entry = FALSE
+	. += ")"
+
+//Converts a list of turfs into TGM file format
+/proc/write_map(minx as num, \
+				miny as num, \
+				minz as num, \
+				maxx as num, \
+				maxy as num, \
+				maxz as num, \
+				save_flag = SAVE_ALL, \
+				shuttle_area_flag = SAVE_SHUTTLEAREA_DONTCARE, \
+				list/obj_blacklist = list())
```

Comment:
we probably don't need the `as num` here, just doc that they're integers. also this multilining sucks and should be like this:

```suggestion
/proc/write_map(
	minx as num,
	miny as num,
	minz as num,
	maxx as num,
	maxy as num,
	maxz as num,
	save_flag = SAVE_ALL,
	shuttle_area_flag = SAVE_SHUTTLEAREA_DONTCARE,
	list/obj_blacklist = list(),
)

we don't need the backslashes in byond-level list stuff like this
```